From 6a88b4a0325d6fa0a7f2c635f163efeb6a747888 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Sun, 12 Apr 2026 16:18:45 +0800
Subject: [PATCH 045/243] Kbuild: Enable LLVM machine pipeliner

This is the LLVM equivalent to GCC's swing modulo scheduling. Enable it
as well to get the similar optimizations when building with clang.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 Makefile                      | 1 +
 drivers/acpi/processor_idle.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 543115a4b5ca..804dea13c40a 100644
--- a/Makefile
+++ b/Makefile
@@ -941,6 +941,7 @@ endif
 # This pass looks at innermost loops and reorders their instructions by
 # overlapping different iterations.
 KBUILD_CFLAGS += $(call cc-option,-fmodulo-sched -fmodulo-sched-allow-regmoves -fivopts)
+KBUILD_CFLAGS += $(call cc-option,-mllvm -enable-pipeliner)
 
 # Always set `debug-assertions` and `overflow-checks` because their default
 # depends on `opt-level` and `debug-assertions`, respectively.
diff --git a/drivers/acpi/processor_idle.c b/drivers/acpi/processor_idle.c
index 9c93e8858024..4b9cbcf0d1c9 100644
--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -1427,6 +1427,7 @@ void acpi_processor_power_init(struct acpi_processor *pr)
 	if (disabled_by_idle_boot_param())
 		return;
 
+#ifdef CONFIG_X86
 	/*
 	 * The bus master check on AMD is observed to be very costly and results
 	 * in not only increased power consumption, but also lots of CPU cycles
@@ -1437,6 +1438,7 @@ void acpi_processor_power_init(struct acpi_processor *pr)
 		bm_check_disable = true;
 		pr_info("Enabled processor.bm_check_disable for AMD cpuidle\n");
 	}
+#endif
 
 	if (!acpi_processor_get_power_info(pr))
 		pr->flags.power_setup_done = 1;
-- 
2.55.0

